### PR TITLE
Error on unknown parameter name in parameter conditional type

### PR DESCRIPTION
**Pull request by rvanvelzen**
Fix phpstan/phpstan#8284

### CODE REVIEW

File: tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php [hunk 1]
```diff
@@ -70,4 +70,14 @@ public function testRule(): void
 		]);
 	}
 
+	public function testBug8284(): void
+	{
+		$this->analyse([__DIR__ . '/data/bug-8284.php'], [
+			[
+				'Conditional return type references unknown parameter $callable.',
```
Comment (mad-briller) [inline tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php:77]: there is no `@return` tag on the `replaceCallback()` method so is this error message a bit misleading?
it looks like it's really a conditional parameter type?

Comment (ondrejmirtes) [reply tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php:77]: The feature is actually "conditonal types", not "conditional return types" - it was implemented as a first thing after we released "conditional return types".

Comment (ondrejmirtes) [reply tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php:77]: But for the sake of backward compatibility, let's leave it like this.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**